src/sas/qtgui/Perspectives/Inversion/InversionPerspective.py

```diff
@@ -12,135 +14,213 @@
+        # Remove the previous last tab
+        self.tabCloses(index)
+
+    def tabCloses(self, index):
+        """
+        Update local bookkeeping on tab close
+        """
+        # don't remove the last tab
+        if len(self.tabs) <= 1:
+            return
+        self.closeTabByIndex(index)
+
+    def closeTabByIndex(self, index):
+        """
+        Close/delete a tab with the given index.
+        No checks on validity of the index.
+        """
+        try:
+            self.removeTab(index)
+            del self.tabs[index]
+
+        except IndexError:
+            # The tab might have already been deleted previously
+            pass
+
+
+    def closeTabByName(self, tab_name):
+        """
+        Given name of the tab - close it
+        """
+        for tab_index in range(len(self.tabs)):
+            if self.tabText(tab_index) == tab_name:
+                self.tabCloses(tab_index)
+        pass # debug hook
+    ######################################################################
+
+    def serializeAll(self):
+        """
+        Serialize the inversion state so data can be saved
+        serialize all active inversion pages and return
+        a dictionary: {data-id: {self.name: {inversion-state}}}
+        """
+        state = {}
+        index = [tab.tab_id for tab in self.tabs]
+        try:          
+            for i in index:              
+                state.update(self.getSerializePage(i))
+        except TypeError:
+            return
+        return state
+
+    def serializeCurrentPage(self):
+        # serialize current (active) page
+        return self.getSerializePage(self.currentIndex())
+
+    def getSerializePage(self, index=None):
+        """
+        Serialize and return a dictionary of {data_id: inversion-state}
+        Return original dictionary if no data
+        """
+        state = {}
+        if index is None:
+            index = self.currentIndex()
+        # If data on tab empty - do nothing
+        try:
+            tab = self.tabs[index]        
+            if tab.logic.data_is_loaded:
+                tab_data = tab.getPage()
+                data_id = tab_data.pop('data_id', '')
+                state[data_id] = {'pr_params': tab_data}
+            return state
+        except:
+             return
+            
+
+    def _calculateUpdate(self, params: dict, tab_name):
+        """ Update the perspective using a dictionary of parameters
+        e.g. those loaded via open project or open analysis menu items"""
+        raise NotImplementedError("Update from parameters not implemented yet.")
+
 
     ######################################################################
     # Base Perspective Class Definitions
 
+
+    @property
+    def supports_reports(self):
+        return True
+        
+    @property
+    def supports_fitting(self):
+        return False
+
     def communicator(self):
-        return self.communicate
+        return self.filesWidget.communicate
```

Comment:
there's only one instance of `Communicate` - created in `GuiManager`. Both `self.communicate` (from the parent, which is GuiManager) or from `filesWidget` should be the same object. Why change it here?